qt/widgets/common/src/AlgorithmRunner.cpp

```diff
@@ -1,120 +1,47 @@
-
-//----------------------------------------------------------------------------------------------
-/** Destructor
- */
-AlgorithmRunner::~AlgorithmRunner() {
-  if (m_asyncAlg) {
-    m_asyncAlg->removeObserver(m_finishedObserver);
-    m_asyncAlg->removeObserver(m_errorObserver);
-    m_asyncAlg->removeObserver(m_progressObserver);
-  }
-  delete m_asyncResult;
-}
-
-//--------------------------------------------------------------------------------------
-/** If an algorithm is already running, cancel it.
- * Does nothing if no algorithm is running. This blocks
- * for up to 1 second to wait for the algorithm to finish cancelling.
- */
-void AlgorithmRunner::cancelRunningAlgorithm() {
-  // Cancel any currently running algorithms
-  if (m_asyncAlg) {
-    if (m_asyncAlg->isRunning()) {
-      m_asyncAlg->cancel();
-    }
-    if (m_asyncResult) {
-      m_asyncResult->tryWait(1000);
-      delete m_asyncResult;
-      m_asyncResult = nullptr;
-    }
-    m_asyncAlg->removeObserver(m_finishedObserver);
-    m_asyncAlg->removeObserver(m_errorObserver);
-    m_asyncAlg->removeObserver(m_progressObserver);
-    m_asyncAlg.reset();
-  }
+AlgorithmRunner::AlgorithmRunner(std::unique_ptr<IJobRunner> jobRunner)
+    : m_jobRunner(std::move(jobRunner)), m_subscriber(), m_lastAlgorithm() {
+  m_jobRunner->subscribe(this);
 }
 
-//--------------------------------------------------------------------------------------
-/** Begin asynchronous execution of an algorithm and observe its execution
- *
- * @param alg :: algorithm to execute. All properties should have been set
- *properly.
- */
-void AlgorithmRunner::startAlgorithm(Mantid::API::IAlgorithm_sptr alg) {
-  if (!alg)
-    throw std::invalid_argument("AlgorithmRunner::startAlgorithm() given a NULL Algorithm");
-  if (!alg->isInitialized())
-    throw std::invalid_argument("AlgorithmRunner::startAlgorithm() given an uninitialized Algorithm");
+void AlgorithmRunner::subscribe(IAlgorithmRunnerSubscriber *subscriber) { m_subscriber = subscriber; }
 
-  cancelRunningAlgorithm();
+void AlgorithmRunner::execute(IConfiguredAlgorithm_sptr algorithm) {
+  m_jobRunner->executeAlgorithm(std::move(algorithm));
+}
 
-  // Observe the algorithm
-  alg->addObserver(m_finishedObserver);
-  alg->addObserver(m_errorObserver);
-  alg->addObserver(m_progressObserver);
-  // Start asynchronous execution
-  m_asyncAlg = alg;
-  m_asyncResult = new Poco::ActiveResult<bool>(m_asyncAlg->executeAsync());
+void AlgorithmRunner::execute(std::deque<IConfiguredAlgorithm_sptr> algorithmQueue) {
+  m_jobRunner->setAlgorithmQueue(algorithmQueue);
+  m_jobRunner->executeAlgorithmQueue();
 }
 
-/// Get back a pointer to the running algorithm
-Mantid::API::IAlgorithm_sptr AlgorithmRunner::getAlgorithm() const { return m_asyncAlg; }
+void AlgorithmRunner::notifyBatchComplete(bool error) { m_subscriber->notifyBatchComplete(m_lastAlgorithm, error); };
```

Comment:
extra semi-colon

Reply:
Removed in last commit of this https://github.com/mantidproject/mantid/pull/37255